go-controller/pkg/util/ovs.go

```diff
@@ -895,3 +907,50 @@ func GetOVSPortPodInfo(hostIfName string) (bool, string, string, error) {
 	}
 	return true, sandbox, nadName, nil
 }
+
+type TunnelInfo struct {
+	Name       string
+	LocalIP    string
+	RemoteIP   string
+	RemoteName string
+}
+
+func GetGeneveTunnels() ([]TunnelInfo, error) {
+	var geneveTunnels []TunnelInfo
+	stdout, stderr, err := RunOVSVsctl("--format=json", "--columns=name,options",
+		"find", "Interface", "type=geneve")
+	if err != nil {
+		return geneveTunnels, fmt.Errorf("failed to retrieve Geneve tunnels stderr %v: %v", stderr, err)
+	}
+	if stdout == "" {
+		return geneveTunnels, nil
+	}
+
+	var result map[string]interface{}
+	err = json.Unmarshal([]byte(stdout), &result)
+	if err != nil {
+		return geneveTunnels, fmt.Errorf("failed to parse json output for Geneve tunnels, err: %v", err)
+	}
+	// sample output
+	// {"data":[["ovn-203886-0",["map",[["csum","true"],["key","flow"],["local_ip","10.89.0.3"],["remote_ip","10.89.0.4"],["remote_name","203886dd-d6e1-4b48-bdd9-0a6bbfc6acdc"]]]],
+	//          ["ovn-4e658c-0",["map",[["csum","true"],["key","flow"],["local_ip","10.89.0.3"],["remote_ip","10.89.0.2"],["remote_name","4e658c65-a43f-4bb9-a36f-723ef1d96a66"]]]]],
+	//  "headings":["name","options"]}
+	data := result["data"].([]interface{})
+	for _, item := range data {
+		entry := item.([]interface{})
+		name := entry[0].(string)
```

Comment:
This code scares me with all the casting

Reply:
look at the new function, hope that is fine with you.